hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -1131,6 +1155,8 @@ private void verifyCapacityAndClasses(long capacitySize, String ioclass, String
   }
 
   private void parsePB(BucketCacheProtos.BucketCacheEntry proto) throws IOException {
+    ((PersistentIOEngine) ioEngine).verifyFileIntegrity(proto.getChecksum().toByteArray(),
```

Comment:
For compatibility, u need to check whether the checksum is there or not. Use proto.hasChecksum() before calling the getter.